### ISSUE TITLE
Play's hot reload loads the server TWICE for the first run of "runPlayBinary"

### ISSUE THREAD

**Issue by ymwang1984** [gradle/gradle#8878]
This issue has been extensively discussed in the below forum question.

https://discuss.gradle.org/t/why-does-plays-runplaybinary-always-load-twice/28158

I've filed an original issue in https://github.com/gradle/gradle/issues/6789

per @lacasseio , I may have to file a separate one.

### Expected Behavior

When we start the Play application, we expect it to load just once before starting to listen for requests.

### Current Behavior

However, in reality it loads **_twice_**.

Per @bigdaz from Gradle Inc in the forum question

> The fact that we execute the build twice on initial startup is a bug.

### Context

Server start up becomes slower, which impacts developer productivity. "runPlayBinary" is invoked by more than 4000 times per day in the company. Even thought the second run could be just checking task input and output and marking every step as UP-TO-DATE, it still takes time for large scale applications.

### Steps to Reproduce 

This can be easily reproduced in sample Play apps that are inside Gradle.

> [info] p.c.s.AkkaHttpServer - Listening for HTTP on /0:0:0:0:0:0:0:0:9000
>  
> > Task :runPlayBinary
> Running Play App (:runPlayBinary) at http://localhost:9000/
>  
> BUILD SUCCESSFUL in 54s
> 7 actionable tasks: 7 executed
>  
> Reloadable deployment detected. Entering continuous build.
>  
> BUILD SUCCESSFUL in 54s
> 7 actionable tasks: 1 executed, 6 up-to-date
>  
> Waiting for changes to input files of tasks... (ctrl-d to exit)

### Your Environment

Build scan URL:

It happens both on Mac and Linux.

**Comment by victorbr** [comment]
I observe the same issue:
```
10:33:29: Executing task 'runPlay'...

Starting Gradle Daemon...
Gradle Daemon started in 4 s 340 ms
> Task :minifyPlayJavaScript NO-SOURCE
> Task :createPlayAssetsJar UP-TO-DATE
> Task :compileKotlin UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :compilePlayTwirlTemplates UP-TO-DATE
> Task :compileScala UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :inspectClassesForKotlinIC
> Task :jar
--- (Running the application, auto-reloading is enabled) ---

< here goes the startup log of our application, redacted>
...
[info] - play.api.Play - Application started (Dev)
[warn] - play.core.server.ssl.DefaultSSLEngineProvider - Using generated key with self signed certificate for HTTPS. This should not be used in production.
[info] - play.core.server.AkkaHttpServer - Listening for HTTP on /0:0:0:0:0:0:0:0:9000
[info] - play.core.server.AkkaHttpServer - Listening for HTTPS on /0:0:0:0:0:0:0:0:9443

> Task :runPlay
Running Play App (:runPlay) at http://localhost:9000/

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.5.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 1m 34s
8 actionable tasks: 3 executed, 5 up-to-date

Reloadable deployment detected. Entering continuous build.

> Task :minifyPlayJavaScript NO-SOURCE
> Task :createPlayAssetsJar UP-TO-DATE
> Task :compileKotlin UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :compilePlayTwirlTemplates UP-TO-DATE
> Task :compileScala UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :inspectClassesForKotlinIC UP-TO-DATE
> Task :jar UP-TO-DATE
> Task :runPlay

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.5.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 4s
8 actionable tasks: 1 executed, 7 up-to-date

Waiting for changes to input files of tasks... (ctrl-d to exit)

--- (RELOAD) ---

< log lines of server shut down and 2nd startup >
...
[info] - play.api.Play - Application started (Dev)

```

**Comment by victorbr** [comment]
Hmmm... This is something elusive. Before I posted it kept happening all the time. And then I tried to debug it, played with Gradle options. At some point it stopped happening, then I returned everything to where it was but the problem disappeared and for the moment I couldn't reproduce it anymore! 🤷‍♂️

**Comment by Astro03** [comment]
@victorbr  maybe try rebooting and this will happen again?

**Comment by bmuskalla** [comment]
The plugins are superseded by https://gradle.github.io/playframework/ and will be [removed in one of the next major releases](https://github.com/gradle/gradle/issues/9434).

**Comment by Astro03** [comment]
@bmuskalla  Shouldn't this ticket still be open until AFTER the release?

**Comment by bmuskalla** [comment]
We don't plan to fix this before the removal of the plugin. As I said above, please refer to the new plugin for play development.

**Comment by victorbr** [comment]
I have observed this problem with 
```
id("org.gradle.playframework") version "0.9"
```
Is this the "new" plugin?

**Comment by bmuskalla** [comment]
Yes, that's the external plugin at https://github.com/gradle/playframework - it seems the original author already posted this issue [on there as well](https://github.com/gradle/playframework/issues/86).

**Comment by Astro03** [comment]
@bmuskalla  Just to clarify. Are you saying that 

1) The issue is NOT with Gradle but with the play framework plugin?

AND

2) Even when you remove the play plugins from the gradle project (as specified by the link above), the issue will persist since the play project has not fixed this bug?

**Comment by bmuskalla** [comment]
I'm saying that we don't have plans to fix this issue in the old play plugin. Whatever the solution will be in the new plugin to handle this behavior, it may or may not require work in Gradle itself. But that is up to how the new play plugin decides to handle the issue linked above.

**Comment by ldaley** [comment]
FWIW, we see this with other reloadable toolchains relying on continuous build. I'm not sure it's specific to Play.
